makefile

```diff
@@ -677,11 +677,17 @@ install:
 	cp -f $(ESMF_MODDIR)/*.mod $(ESMF_INSTALL_MODDIR_ABSPATH)
 	mkdir -p $(ESMF_INSTALL_LIBDIR_ABSPATH)
 	cp -f $(ESMF_LIBDIR)/libesmf*.* $(ESMF_INSTALL_LIBDIR_ABSPATH)
-	@for lib in $(wildcard $(ESMF_INSTALL_LIBDIR_ABSPATH)/libesmf*.dylib) foo ; do \
+ifneq (,$(filter $(ESMF_OS),Darwin DARWIN darwin))
+	@for lib in $(ESMF_INSTALL_LIBDIR_ABSPATH)/libesmf*.dylib foo ; do \
 	  if [ $$lib != "foo" ]; then \
 	    install_name_tool -id "$$lib" $$lib ; \
 	  fi ; \
+	  if [ $$lib == "$(ESMF_INSTALL_LIBDIR_ABSPATH)/libesmftrace_preload.dylib" ]; then \
+	    install_name_tool -change "$(ESMF_LIBDIR)/libesmf.dylib" \
```

Comment:
I'm probably missing something here -- why is install_name_tool making a change to libesmf.dylib if the loop is on libesmftrace_preload.dylib?

Reply:
libesmftrace_preload.dylib references libesmf.dylib. The references is static when the file is moved to the installation directory. install_name_tool -change let's you change the path to libesmf.dylib in libesmftrace_preload.dylib

```
otool -L libesmftrace_preload.dylib
libesmftrace_preload.dylib:
	<some_path>/Utils/esmf/O/dev_gnu-9.4.0/lib/libesmftrace_preload.dylib (compatibility version 0.0.0, current version 0.0.0)
	<some_path>/Utils/openmpi/3.0.0_gnu-9.4.0/lib/libmpi_mpifh.40.dylib (compatibility version 41.0.0, current version 41.0.0)
	/usr/local/opt/gcc@9/lib/gcc/9/libgfortran.5.dylib (compatibility version 6.0.0, current version 6.0.0)
	<some_path>/Utils/netcdf/4.7.0_gnu-9.4.0/lib/libnetcdff.6.dylib (compatibility version 9.0.0, current version 9.1.0)
	<some_path>/Utils/netcdf/4.7.0_gnu-9.4.0/lib/libnetcdf.15.dylib (compatibility version 16.0.0, current version 16.1.0)
	<some_path>/Utils/esmf/O/dev_gnu-9.4.0/lib/libesmf.dylib (compatibility version 0.0.0, current version 0.0.0)
	<some_path>/Utils/openmpi/3.0.0_gnu-9.4.0/lib/libmpi.40.dylib (compatibility version 41.0.0, current version 41.0.0)
	/usr/local/opt/gcc@9/lib/gcc/9/libstdc++.6.dylib (compatibility version 7.0.0, current version 7.28.0)
	/usr/local/opt/gcc@9/lib/gcc/9/libgomp.1.dylib (compatibility version 2.0.0, current version 2.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1281.100.1)
	/usr/local/lib/gcc/9/libgcc_s.1.dylib (compatibility version 1.0.0, current version 1.0.0)
```